### PR TITLE
<main> Convert test_wl_cronjob and test_wl_statefulset test to go

### PR DESCRIPTION
**Pull request by lscalabrini01**
Implemented TestWorkloadCronjob test
Implemented TestWorkloadStatefulset test

Backports

[v2.9 ](https://github.com/rancher/rancher/pull/46717)
[v2.8](https://github.com/rancher/rancher/pull/46716)
[v2.7](https://github.com/rancher/rancher/pull/46725)

### CODE REVIEW

**Review by floatingman — DISMISSED** [review]
LGTM

File: tests/v2/actions/workloads/statefulset/statefulset.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	statefulsetResource := dynamicClient.Resource(StatefulsetGroupVersionResource).Namespace(namespaceName)
+
+	unstructuredResp, err := statefulsetResource.Create(context.TODO(), unstruc.MustToUnstructured(statefulsetTemplate), metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = charts.WatchAndWaitStatefulSets(client, clusterID, namespaceName, metav1.ListOptions{
```
Comment (slickwarren) [inline tests/v2/actions/workloads/statefulset/statefulset.go:60]: usually in our create functions, we don't include the wait. But I see for daemonset.go -> create, it was already there. What do you think, should it be included or done in the test itself? i.e. in the `TestWorkloadSideKick` function, you updated the line to watchAndWait pods.

**Review by slickwarren — COMMENTED** [review]
lgtm, just a couple things that might be worth adding.

File: tests/v2/validation/workloads/workload_test.go [hunk 2]
```diff
@@ -106,6 +108,68 @@ func (w *WorkloadTestSuite) TestWorkloadDaemonSet() {
...
+		nil,
+	)
+	podTemplate := workloads.NewPodTemplate(
+		[]corev1.Container{containerTemplate},
+		[]corev1.Volume{},
+		[]corev1.LocalObjectReference{},
+		nil,
+	)
+
+	_, err = cronjob.CreateCronjob(w.client, w.cluster.ID, namespace.Name, "*/1 * * * *", podTemplate)
```
[the remark this answers is not in the document]
Comment (lscalabrini01) [reply tests/v2/validation/workloads/workload_test.go:138]: Hi @slickwarren I am validating it on the CreateCronjob function on this [line](https://github.com/lscalabrini01/rancher/blob/qa-issue-1441-test_wl_cronjob_statefulset-main/tests/v2/actions/workloads/cronjob/cronjob.go#L82)

**Review by slickwarren — DISMISSED** [review]

**Review by floatingman — APPROVED** [review]
LGTM

**Comment by igomez06** [comment]
Let's wait until https://github.com/rancher/rancher/pull/46662 is merged first.

**Review by anupama2501 — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]
lgtm

**Pull request merged**